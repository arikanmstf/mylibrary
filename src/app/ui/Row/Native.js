// @flow
import React from 'react';
import { StyleSheet, View } from 'react-native';
import type { RowProps } from './types';

const defaultStyle = {
  marginTop: 5,
  marginBottom: 5,
  flexDirection: 'column',
};

const Row = (props: RowProps) => {
  const { style, fullHeight, ...other } = props;
  const heightStyle = fullHeight ? { height: '100%' } : {};
  const mergedStyles = StyleSheet.create({
    container: {
      ...defaultStyle,
      ...heightStyle,
      ...style,
    },
  });

  return (
    <View
      style={mergedStyles.container}
      {...other}
    />
  );
};

export default Row;
