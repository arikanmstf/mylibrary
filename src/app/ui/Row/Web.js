/**
 * Web Component Template By => create-module script
 * @version 1.0.0
 *
 */

// @flow
import styled from 'styled-components';

const Row = styled.div`
margin: 5px 0;
display: flex;
`;

export default Row;
