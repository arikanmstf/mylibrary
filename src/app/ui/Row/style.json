{
  "marginTop": 5,
  "marginBottom": 5
}
