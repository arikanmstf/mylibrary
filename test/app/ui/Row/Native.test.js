/**
 * Native Component Test Template By => create-module script
 * @version 1.0.0
 *
 */

import React from 'react';
import { shallow } from 'enzyme';
import Row from 'ui/Row/Native';

describe('test/app/ui/Row/Native.test.js', () => {
  it('Render', () => {
    const props = {};
    const wrapper = shallow(<Row {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
