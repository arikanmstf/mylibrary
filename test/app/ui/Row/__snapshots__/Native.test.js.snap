// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test/app/ui/Row/Native.test.js Render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Row />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "style": Object {
        "flexDirection": "column",
        "marginBottom": 5,
        "marginTop": 5,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "style": Object {
          "flexDirection": "column",
          "marginBottom": 5,
          "marginTop": 5,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
